src/main.rs

```diff
@@ -32,78 +46,84 @@ fn main() -> Result<(), Box<dyn Error>> {
     // Render loop in a separate thread
     let (render_tx, render_rx) = mpsc::channel();
     let render_handle = thread::spawn(move || {
-        let mut last_frame = frame::new_frame();
-        let mut stdout = io::stdout();
-        render::render(&mut stdout, &last_frame, &last_frame, true);
-        loop {
-            let curr_frame = match render_rx.recv() {
-                Ok(x) => x,
-                Err(_) => break,
-            };
-            render::render(&mut stdout, &last_frame, &curr_frame, false);
-            last_frame = curr_frame;
-        }
+        render_screen(render_rx);
     });
 
     // Game loop
     let mut player = Player::new();
     let mut instant = Instant::now();
     let mut invaders = Invaders::new();
     let mut score = Score::new();
+    let mut menu = Menu::new();
+    let mut scene = Scenes::Menu;
+
+    fn reset_game(scene: &mut Scenes, player: &mut Player, invaders: &mut Invaders) {
+        *scene = Scenes::Menu;
+        *player = Player::new();
+        *invaders = Invaders::new();
+    }
+
+    fn draw(drawables: Vec<&dyn Drawable>, frame: &mut Frame) {
+        for drawable in drawables {
+            drawable.draw(frame);
+        }
+    }
 
-    'gameloop: loop {
+    while scene != Scenes::End {
         // Per-frame init
         let delta = instant.elapsed();
         instant = Instant::now();
         let mut curr_frame = new_frame();
 
-        // Input
-        while event::poll(Duration::default())? {
-            if let Event::Key(key_event) = event::read()? {
-                match key_event.code {
-                    KeyCode::Left => player.move_left(),
-                    KeyCode::Right => player.move_right(),
-                    KeyCode::Char(' ') | KeyCode::Enter => {
-                        if player.shoot() {
-                            audio.play("pew");
-                        }
-                    }
-                    KeyCode::Esc | KeyCode::Char('q') => {
-                        audio.play("lose");
-                        break 'gameloop;
-                    }
-                    _ => {}
-                }
+        if scene == Scenes::Menu {
+            // How can I improve this? (used to avoid problems with references)
```

Comment:
If we kept this code, the answer would be to stop using references!  You should avoid using references with `Copy` types whenever possible.